src/main/java/com/commercetools/paymenttoorderprocessor/jobs/actions/MessageReader.java

```diff
@@ -127,7 +128,7 @@ private void fetchUnprocessedMessagesFromPlatform() {
     private MessageQuery buildQuery() {
 
         MessageQuery messageQuery = MessageQuery.of()
-                .withPredicates(m -> m.type().is(MESSAGETYPE))
+                .plusPredicates(m -> m.type().is(PAYMENT_TRANSACTION_ADDED).or(m.type().is(PAYMENT_TRANSACTION_STATE_CHANGED)))
```

Comment:
as it goes same .in() might be better for best practices. But you could skip that comment as those almost same.

Reply:
I tried and get this error:
```
io.sphere.sdk.client.ErrorResponseException: detailMessage: Malformed parameter: where: Syntax error while parsing 'where'. Invalid input 'P', expected input parameter or primitive value (line 1, column 10):
type in (PaymentTransactionAdded, PaymentTransactionStateChanged)
         ^
summary: GET https://api.europe-west1.gcp.commercetools.com/lam-sphere-node-sdk-test/messages?where=type+in+%28PaymentTransactionAdded%2C+PaymentTransactionStateChanged%29&where=lastModifiedAt%3E%222021-06-10T13%3A36%3A20.694Z%22&sort=lastModifiedAt+asc&limit=500&offset=0 failed  with response code 400 with X-Correlation-ID `lam-sphere-node-sdk-test/3b94f71c-08be-4990-a035-47bb82b8730b` on 2021-06-10T13:41:13.092Z
http response formatted body: {
  "statusCode" : 400,
  "message" : "Malformed parameter: where: Syntax error while parsing 'where'. Invalid input 'P', expected input parameter or primitive value (line 1, column 10):\ntype in (PaymentTransactionAdded, PaymentTransactionStateChanged)\n         ^",
  "errors" : [ {
    "code" : "InvalidInput",
    "message" : "Malformed parameter: where: Syntax error while parsing 'where'. Invalid input 'P', expected input parameter or primitive value (line 1, column 10):\ntype in (PaymentTransactionAdded, PaymentTransactionStateChanged)\n         ^"
  } ]
}
```

Is there sth missing in my change? https://github.com/commercetools/commercetools-payment-to-order-processor/pull/71/commits/6099482050d0ee11208a71c1c50fb1fee2a520b9

Reply:
decoded url looks like: `where=type in (PaymentTransactionAdded, PaymentTransactionStateChanged)...` maybe it should be a string not sure why it's like that.

Reply:
Seems like a bug to me. I cannot pass String to `isIn` method as it requires Iterable. I will keep it as it is and create a github issue for the JVM SDK.

Reply:
is it using the latest version of SDK ?

Reply:
Yes, it's 1.63.0: https://github.com/commercetools/commercetools-jvm-sdk/releases/

Reply:
Did you try as shown here: https://github.com/commercetools/commercetools-jvm-sdk/blob/2fcacb7c39162aa5c9dc9e9e57d04ec337792e2c/commercetools-internal-docs/src/test/java/io/sphere/sdk/meta/QueryDocumentationTest.java#L36

Reply:
Also when I look at the decoded query:
```
where=type in (PaymentTransactionAdded, PaymentTransactionStateChanged)&where=lastModifiedAt>"2021-06-10T13:36:20.694Z"
```
There should be only single `where` statement and both conditions linked with an AND, right? Also, I think it might be preferable to have `lastModifiedAt` condition as the very first part of the query.

Reply:
> Also when I look at the decoded query:
> 
> ```
> where=type in (PaymentTransactionAdded, PaymentTransactionStateChanged)&where=lastModifiedAt>"2021-06-10T13:36:20.694Z"
> ```
> 
> There should be only single `where` statement and both conditions linked with an AND, right? Also, I think it might be preferable to have `lastModifiedAt` condition as the very first part of the query.

I checked and in our code we use `.plusPredicates()` method to add `lastModifiedAt` to the existing query. And in the JVM SDK code, for every predicate we join using `WHERE`: https://github.com/commercetools/commercetools-jvm-sdk/blob/2fcacb7c39162aa5c9dc9e9e57d04ec337792e2c/commercetools-sdk-base/src/main/java/io/sphere/sdk/queries/MetaModelQueryDslImpl.java#L290. This apparently works as we use `.plusPredicates()` in other libs as well.

As to having `lastModifiedAt` as the first part - I'm not sure if it would help or not. I remembered discussing this kind of query optimization in another PR. It could be beneficial, it could also make it worse, so I would not change it for now. Optimization should be done only when you have problems, and so far I don't see any complaints about the performance of this job.

Reply:
https://github.com/commercetools/commercetools-payment-to-order-processor/pull/71/commits/45ae24b740436ffe870100e85aae86336ef1885f